cmd/kube-apiserver/app/server.go

```diff
@@ -569,7 +575,43 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 	}
 
 	if s.Etcd.EncryptionProviderConfigFilepath != "" {
-		transformerOverrides, err := encryptionconfig.GetTransformerOverrides(s.Etcd.EncryptionProviderConfigFilepath)
+		// Obtain GCE cloud for a Google Cloud KMS connection if needed.
+		// gce package cannot be accessed inside staging, which is why this function is passed from here.
+		getGCECloud := func() (*kms.GCECloud, error) {
+			cloud, err := cloudprovider.InitCloudProvider(s.CloudProvider.CloudProvider, s.CloudProvider.CloudConfigFile)
+			if err != nil {
+				return nil, err
+			}
+
+			var cloudkmsService *cloudkms.Service
+			var projectID string
+
+			// Safe when cloud is nil too.
```

Comment:
It means that the check will not throw an exception if cloud is nil. It was not obvious for me until I read the documentation for type assertions. Will make it more clear.